File: f312.cpp
#include <iostream>
using namespace std;

int main(void) {
    int A1, B1, C1, A2, B2, C2, n, temp, MAX = -9999;
    cin >> A1 >> B1 >> C1 >> A2 >> B2 >> C2 >> n;
    for (int i = 0; i <= n; i++) {
        temp = (A1*i*i + B1*i + C1) + (A2*(n-i)*(n-i) + B2*(n-i) + C2);
        if (temp > MAX) {
            MAX = temp;
        }
    }
    cout << MAX << endl;
    return 0;
}

File: f312.py
# -*- coding: utf-8 -*-

A1, B1, C1 = list(map(int, input().split()))
A2, B2, C2 = list(map(int, input().split()))
n = int(input())
MAX = -999
for i in range(n+1):
    temp = (A1*i*i + B1*i + C1) + (A2*(n-i)*(n-i) + B2*(n-i) + C2)
    if (temp > MAX):
        MAX = temp
print(MAX)    